src/librustdoc/json/conversions.rs

```diff
@@ -127,7 +127,7 @@ impl FromWithTcx<clean::GenericArgs> for GenericArgs {
             },
             Parenthesized { inputs, output } => GenericArgs::Parenthesized {
                 inputs: inputs.into_iter().map(|a| a.into_tcx(tcx)).collect(),
-                output: output.map(|a| a.into_tcx(tcx)),
+                output: output.map(|a| (*a).into_tcx(tcx)),
```

Comment:
nit: this could be simpler:
```suggestion
                output: output.map(|&a| a.into_tcx(tcx)),
```
In general I don't know that the IntoTcx trait pulls its weight, but it certainly shouldn't block this PR.

Reply:
Are you sure `&` patterns work with `Box`? They don't seem to on the Playground. I don't think Deref patterns have been implemented yet.

Reply:
Yeah, it doesn't seem to work:

```
error[E0308]: mismatched types
   --> src/librustdoc/json/conversions.rs:130:37
    |
130 |                 output: output.map(|&a| a.into_tcx(tcx)),
    |                                     ^--
    |                                     ||
    |                                     |expected due to this
    |                                     expected struct `std::boxed::Box`, found reference
    |                                     help: did you mean `a`: `&std::boxed::Box<types::Type>`
    |
    = note: expected struct `std::boxed::Box<types::Type>`
            found reference `&_`
```

Reply:
Oh, I forgot this was a smart pointer. I think `box a` would work, but IDK how much simpler it is at that point.

Reply:
Yeah, I think it would work, but it seems more complex in some ways. I'll just leave it as `(*x)` for now.